src/util/Repeat.hpp

```diff
@@ -43,9 +43,11 @@ class Repeat {
     /**
      * @brief Construct a new Repeat object
      *
-     * @param ioc The io_context to use
+     * @param ctx The io_context-like object to use
      */
-    Repeat(boost::asio::io_context& ioc);
+    Repeat(auto& ctx) : timer_(ctx)
```

Comment:
This makes it usable with other types of contexts such as `thread_pool` or `strand`.

Reply:
Maybe worth to leave this comment in the code.